src/main/java/thaumicenergistics/client/gui/GuiEssentiaStorageBus.java

```diff
@@ -232,12 +234,19 @@ protected void mouseClicked( final int mouseX, final int mouseY, final int mouse
 		// Call super
 		super.mouseClicked( mouseX, mouseY, mouseButton );
 
+		ItemStack temp = null;
+		if( draggedStack != null )
+		{
+			temp = draggedStack.copy();
+			draggedStack = null;
+		}
+
 		for( WidgetAspectSlot aspectSlot : this.aspectWidgetList )
 		{
 			if( aspectSlot.isMouseOverWidget( mouseX, mouseY ) )
 			{
 				// Get the aspect of the currently held item
-				Aspect itemAspect = EssentiaItemContainerHelper.INSTANCE.getFilterAspectFromItem( this.player.inventory.getItemStack() );
+				Aspect itemAspect = EssentiaItemContainerHelper.INSTANCE.getFilterAspectFromItem(temp == null ? this.player.inventory.getItemStack() : temp);
```

Comment:
Variable `temp` is redundant, you can just `getFilterAspectFromItem(draggedStack == null ? this.player.inventory.getItemStack() : draggedStack.copy())` and then `draggedStack = null;`